packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  State,
+} from '@stencil/core';
+
+const DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY = 'DISMISSED_PROMO_BANNERS';
+@Component({
+  tag: 'va-promo-banner',
+  styleUrl: 'va-promo-banner.css',
+  shadow: true,
+})
+export class VaPromoBanner {
+  @Element() el: HTMLElement;
+
+  /**
+   * `href` attribute for the `<a>` tag.
+   */
+  @Prop() href: string;
+
+  /**
+   * `target` attribute for the `<a>` tag.
+   */
+  @Prop() target: string;
+
+  /**
+   * Controls which icon gets used. Options are 'announcement', 'news', or 'email-signup'.
+   * */
+  @Prop() type: 'announcement' | 'news' | 'email-signup';
+
+  /**
+   * Analytics tracking function(s) will not be called
+   */
+  @Prop() disableAnalytics: boolean;
+
+  /**
+   * Fires when the component is closed by clicking on the close icon.
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  closeEvent: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link or the dismiss icon is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * Keep track of locally dismissed Banners
+   * */
+  @State() dismissedBanners: Array<String> = [];
+
+  private closeHandler(): void {
+    this.closeEvent.emit();
+    // Derive the updated dismissed banners based on the id attribute set on the component.
+    const updatedDismissedBanners = [...this.dismissedBanners, this.el.id];
+
+    // Update dismissedBanners in state.
+    this.dismissedBanners = updatedDismissedBanners;
+
+    // Add the promo banner ID to local storage.
+    localStorage.setItem(
+      DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY,
+      JSON.stringify(updatedDismissedBanners),
+    );
+  }
+
+  private handleLinkClick(e: MouseEvent): void {
+    if (!this.disableAnalytics) {
+      const targetEl = e.target as HTMLElement;
+      const targetAnchor =
+        targetEl?.tagName === 'A' ||
+        (targetEl?.tagName === 'SLOT' &&
+          targetEl?.parentElement.tagName === 'A');
+      // If it's a link being clicked, dispatch an analytics event
+      if (targetAnchor) {
+        const detail = {
+          componentName: 'va-promo-banner',
+          action: 'linkClick',
+          details: {
+            text:
+              targetEl.tagName === 'A' ? targetEl.innerText : this.el.innerText,
```

Comment:
Is this leftover from the work on `customRender`? Would the click target here ever not be an `<a>`?

Reply:
Yes just removed it